api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java

```diff
@@ -437,6 +438,25 @@ public Flux<OwnerResponseDTO> getAllOwners() {
                                 .map(addVisitsToOwner(n))
                 );*/
     }
+    @GetMapping(value = "/owners-pagination")
+    public Flux<OwnerResponseDTO> getOwnersByPagination(@RequestParam Optional<Integer> page, @RequestParam Optional<Integer> size) {
+
+        if(page.isEmpty() || page == null){
+            page = Optional.of(0);
```

Comment:
Why is it 0, is it like an array where it starts at 0 ?

Reply:
The pages start at 0, so if no page numbers have been specified, it returns page 0. These pages can be thought of as sections, starting from section zero, with each section containing a certain number of owners. if you look at the back end implementation you will see how it works.